### PR TITLE
added sanity checks

### PR DESCRIPTION
**Pull request by spaul91**
Sanity checks to crash the script if a picard function is requested without specifying its required inputs

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_picardqc/20)

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling and validation for input arguments in the script.
	- Added checks for required variables based on selected Picard functions to prevent execution with insufficient parameters.

- **Bug Fixes**
	- Improved robustness by ensuring appropriate error messages are displayed when required inputs are missing.

### CODE REVIEW

File: src/code.sh [hunk 1]
```diff
@@ -187,6 +187,26 @@ main() {
...
     fi
 
+    # Exit if input args don't align with selected functions
+    if [[ ( "$run_CollectTargetedPcrMetrics" == "true" || \
+            "$run_CollectHsMetrics" == "true" || \
+            "$run_CollectTargetedPcrMetrics" == "true" ) && \
+            ( -z "$sorted_bam" || -z "$fasta_index" || -z "$bedfile" ) ]] ; then
+        err "One of run_CollectTargetedPcrMetrics, run_CollectHsMetrics or run_CollectTargetedPcrMetrics was requested, but one of sorted_bam or fasta_index is missing. Exiting..."
+        exit 1
+    fi
```
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh:197]: Need to fix both of these

File: src/code.sh [hunk 2]
```diff
@@ -187,6 +187,26 @@ main() {
...
+    if [[ "$run_CollectRnaSeqMetrics" == "true" && -z "$sorted_bam" ]]; then
+        err "run_CollectRnaSeqMetrics was requested, but sorted_bam is missing. Exiting..."
+        exit 1
+    fi
+
+    if [[ "$run_CollectVariantCallingMetrics" == "true" && \
+        ( -z "$vcf" || -z "$dbsnp_vcf" ) ]]; then
+        err "run_CollectVariantCallingMetrics was requested, but one of vcf, dbsnp_vcf, or bedfile are missing. Exiting..."
+        exit 1
+    fi
```
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh:208]: need to fix this

File: src/code.sh [hunk 3]
```diff
@@ -187,6 +187,26 @@ main() {
...
+
```
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh:209]: I wish we could do this, but we have to rely on magic variables, so no config to pass

[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh:209]: I read that code wrong. `config` is an associative array. Still, we'd need to set that up ourselves, and it'd make the code look kinda messy IMO

File: src/code.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh]: Done.

File: src/code.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh]: Done.

File: src/code.sh [hunk 4]
```diff
@@ -187,6 +187,26 @@ main() {
...
+        err "run_CollectRnaSeqMetrics was requested, but sorted_bam is missing. Exiting..."
+        exit 1
+    fi
+
+    if [[ "$run_CollectVariantCallingMetrics" == "true" && \
+        ( -z "$vcf" || -z "$dbsnp_vcf" ) ]]; then
+        err "run_CollectVariantCallingMetrics was requested, but one or more of vcf or dbsnp_vcf are missing. Exiting..."
+        exit 1
+    fi
+
```
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh:209]: no thanks

File: src/code.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (spaul91) [reply src/code.sh]: please accept the dismissal

**Review by Yu-jinKim — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_picardqc/20)* status: all files reviewed, 4 unresolved discussions (waiting on @coderabbitai[bot])

**Review by Yu-jinKim — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_picardqc/20)* status: all files reviewed, 4 unresolved discussions (waiting on @coderabbitai[bot] and @spaul91)

**Pull request merged**